src/adapter/src/coord/sequencer.rs

```diff
@@ -3826,6 +3774,97 @@ impl<S: Append + 'static> Coordinator<S> {
 
         Ok(())
     }
+
+    /// Generates the catalog operations to create a linked cluster for the
+    /// source or sink with the given name.
+    pub(crate) async fn create_linked_cluster_ops(
+        &mut self,
+        linked_object_id: GlobalId,
+        name: &QualifiedObjectName,
+        config: &StorageHostConfig,
+    ) -> Result<Vec<catalog::Op>, AdapterError> {
+        let name = self.catalog.resolve_full_name(name, None);
+        let name = format!("{}_{}_{}", name.database, name.schema, name.item);
+        let name = self.catalog.find_available_cluster_name(&name);
+        let arranged_introspection_sources =
+            self.catalog.allocate_arranged_introspection_sources().await;
```

Comment:
Ah, so the reason I allocated these is because if we don't allocate them now, at creation time, we'll have to write a migration to backfill them later. That seemed potentially quite irritating. I figured in the far future, when unification is complete, we'd want linked clusters to be created with the default introspection settings, since those introspection settings will also let you inspect the storage dataflows.

Reply:
That's a good point! As long as we don't allow queries against source/sink-only clusters, nothing can go wrong with these indexes I think, they will just look strange in `mz_indexes`.